shopinvader_schema_sale/models/sale_order_line.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2024 Akretion (https://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    visible_in_shopinvader_api = fields.Boolean(
+        compute="_compute_visible_in_shopinvader_api", store=True
+    )
+
+    def _is_visible_in_shopinvader_api(self):
+        self.ensure_one()
```

Comment:
Using `ensure_one` on internal methods meant to be used in loops is useless IMO.